### DISCUSSION TITLE
Can I apply color to markers on ui.leaflet ?

### DISCUSSION THREAD

**Discussion by kyuhyong** [zauberzeug/nicegui#2355] [Q&A] [answered]
### Question

How to apply color when adding marker on the ui_map created by 
```py
ui_map = ui.leaflet()
ui_map.marker(latlng = (lat,lng))
```

**Comment by falkoschindler** [comment]
Hi @kyuhyong,

It looks like there is no official API for changing the marker color in Leaflet. There are some workarounds though:
https://stackoverflow.com/questions/23567203/leaflet-changing-marker-color

Depending on which approach you want to try (choose from a limited set of colored icons, try to change the color via `hue-rotate`, ...), the workaround for NiceGUI might change. A good starting point is the [`generic_layer`](https://nicegui.io/documentation/leaflet#vector_layers) which accepts generic arguments for whatever object you want to create. But maybe you also need to add some custom CSS via `ui.add_head_html`.

**Comment by kleynjan** [comment, marked as answer]
See #2361, I just added example code there to create markers "under water", bypassing the .marker Nicegui API. The icon there is set through a CSS background-image -- so it can be any shape or color. It **is** a bit more work, though... :smile:

**Reply by thetableman** [reply]
Your code helped me a lot with my project, that's for sharing!

**Comment by cramatt** [comment]
Seems like marker color would be a good API addition for this common use case?

**Reply by falkoschindler** [reply]
Sure. But how would you implement it? If Leaflet doesn't support it and the only way is to add complicated CSS blocks, it doesn't feel like a robust API.

**Reply by evnchn** [reply]
Agreed. As per tradition, if it's a hacky hack, and may break with upstream, they don't get to be in NiceGUI, because a hacky hack is not Nice. 

Similar thing with "Force a smaller-than-dense height for Quasar components": 

https://github.com/zauberzeug/nicegui/wiki/Advanced-operations-and-techniques#force-a-smaller-than-dense-height-for-quasar-components

https://github.com/zauberzeug/nicegui/discussions/4393#discussioncomment-12509420
